Error trying to pull content from remote pull-through cache

**Describe the bug**
Trying to pull content from a remote pull-through cache is failing with:
```
TypeError("int() argument must be a string, a bytes-like object or a number, not 'NoneType'")
```

**To Reproduce**
Steps to reproduce the behavior:
```
curl -uadmin:$PASSWD -X POST $BASE_ADDR/api/pulp/${domain}/api/v3/repositories/npm/npm/ -d '{"name": "test-npm"}' -H 'Content-Type: application/json'
curl -uadmin:$PASSWD  -X POST $BASE_ADDR/api/pulp/${domain}/api/v3/remotes/npm/npm/ -d '{"name": "test-npm", "url": "https://registry.npmjs.org/"}' -H 'Content-Type: application/json'
curl -uadmin:$PASSWD  -X POST $BASE_ADDR/api/pulp/default/api/v3/repositories/npm/npm/0195b9e8-f732-7190-bcde-7584390d52da/sync/ remote=/pulp/default/api/v3/remotes/npm/npm/0195d229-fe90-727d-9c33-1b7c6f51fe2e/
curl -uadmin:$PASSWD -X POST $BASE_ADDR/api/pulp/${domain}/api/v3/distributions/npm/npm/ -d '{"name": "test3", "base_path": "test3", "remote": "/api/pulp/hyagi-s3/api/v3/remotes/npm/npm/0195d292-0a2e-7d01-862d-437ebd748d22/"}' -H 'Content-Type: application/json'
npm install --registry $BASE_ADDR/api/pulp-content/${domain}/test3 react@19.0.0
```

```
[2025-03-26 15:10:01 +0000] [10] [ERROR] Web-handler should return a response instance, got TypeError("int() argument must be a string, a bytes-like object or a number, not 'NoneType'")                                                                                                                                                                                  
Traceback (most recent call last):                                                                                                                                                                                                                                                                                                                                         
  File "/usr/local/lib/pulp/lib64/python3.9/site-packages/aiohttp/web_protocol.py", line 658, in finish_response                                                                                                                                                                                                                                                           
    prepare_meth = resp.prepare                                                                                                                                                                                                                                                                                                                                            
AttributeError: 'TypeError' object has no attribute 'prepare'   
```

**Additional context**

As a workaround, modifying the following lines "fixes" the error:
https://github.com/pulp/pulpcore/blob/6b12bb2ce366dae011838d58e37127a1baefe28b/pulpcore/content/handler.py#L1316-L1318

```
        else:
-            response.headers["X-PULP-ARTIFACT-SIZE"] = str(size)
-            artifacts_size_counter.add(size)
+            if size:
+                response.headers["X-PULP-ARTIFACT-SIZE"] = str(size)
+                artifacts_size_counter.add(size)
```

--- a/CHANGES/6390.bugfix
+++ b/CHANGES/6390.bugfix
@@ -1,0 +1,1 @@
+Set the X-PULP-ARTIFACT-SIZE header at the handle_response_header function.

--- a/pulpcore/content/handler.py
+++ b/pulpcore/content/handler.py
@@ -1248,6 +1248,13 @@ class Handler:
                     response.headers["Content-Range"] = "bytes {0}-{1}/{2}".format(
                         start, stop - 1, content_length
                     )
+
+            content_encoding = headers.get("content-encoding")
+            content_length = headers.get("content-length")
+            if content_length and not content_encoding:
+                response.headers["X-PULP-ARTIFACT-SIZE"] = content_length
+                artifacts_size_counter.add(content_length)
+
             await response.prepare(request)
 
         data_size_handled = 0
@@ -1311,13 +1318,6 @@ class Handler:
                 "affected Pulp Remote, adding a good one and resyncing.\n"
                 "If the problem persists, please contact the Pulp team."
             )
-
-        if content_length := response.headers.get("Content-Length"):
-            response.headers["X-PULP-ARTIFACT-SIZE"] = content_length
-            artifacts_size_counter.add(content_length)
-        else:
-            response.headers["X-PULP-ARTIFACT-SIZE"] = str(size)
-            artifacts_size_counter.add(size)
 
         if save_artifact and remote.policy != Remote.STREAMED:
             content_artifacts = await asyncio.shield(

